bpfman-operator/apis/v1alpha1/shared_types.go

```diff
@@ -0,0 +1,383 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +kubebuilder:validation:Required
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// InterfaceSelector defines interface to attach to.
+// +kubebuilder:validation:MaxProperties=1
+// +kubebuilder:validation:MinProperties=1
+type InterfaceSelector struct {
+	// interfaces refers to a list of network interfaces to attach the BPF
+	// program to.
+	// +optional
+	Interfaces *[]string `json:"interfaces,omitempty"`
+
+	// primaryNodeInterface to attach BPF program to the primary interface on the node. Only 'true' accepted.
+	// +optional
+	PrimaryNodeInterface *bool `json:"primaryNodeInterface,omitempty"`
+}
+
+// ClContainerSelector identifies a set of containers. For example, this can be
+// used to identify a set of containers in which to attach uprobes.
+type ClContainerSelector struct {
+	// namespaces indicate the target namespaces.
+	// +optional
+	// +kubebuilder:default:=""
+	Namespace string `json:"namespace"`
+
+	// pods indicate the target pods. This field must be specified, to select all pods use
+	// standard metav1.LabelSelector semantics and make it empty.
+	Pods metav1.LabelSelector `json:"pods"`
+
+	// containerNames indicate the Name(s) of container(s).  If none are specified, all containers in the
+	// pod are selected.
+	// +optional
+	ContainerNames *[]string `json:"containerNames,omitempty"`
```

Comment:
Any particular reason for this to be `*[]string` over `[]string`?

Also, is there any validation you can put in place on container names? I'm assuming Kubernetes container names fall under one of the requirements specified in https://kubernetes.io/docs/concepts/overview/working-with-objects/names